### PR TITLE
WebGLRenderer.renderPass - making post effects one liners

### PR DESCRIPTION
**Pull request by bhouston**
Prior to this PR it was actually a lot of work to render a post effect.  One needed to create a scene, add a plane to it and an orthographic camera.  One then needed to set a material on that plane and then render it.  It looked like this:

```
// setup
this.camera = new THREE.OrthographicCamera( - 1, 1, 1, - 1, 0, 1 );     
this.quad = new THREE.Mesh( new THREE.PlaneBufferGeometry( 2, 2 ), null );      
this.scene  = new THREE.Scene();        
this.scene.add( this.quad );

// usage:
this.quad.material = this.materialConvolution;
renderer.render( this.scene, this.camera, this.renderTarget, true );
```

This PR makes a function for that common workflow.  This means that rendering post effects is now one line:

```
renderer.renderPass( this.materialConvolution, this.renderTarget, 0x000000, 0 );
```

I find this very useful and I think it will encourage people to use render passes because it makes them easy to use.

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1152,6 +1156,36 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 	}
 
+	this.renderPass = function ( passMaterial, renderTarget, clearColor, clearAlpha ) {
+
+		if( ! this.passScene ) {
```
Comment (dimarudol) [inline src/renderers/WebGLRenderer.js]: I believe here is typo: post/pass

Comment (bhouston) [inline src/renderers/WebGLRenderer.js]: Thanks!  Fixed.

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -71,6 +71,10 @@ THREE.WebGLRenderer = function ( parameters ) {
 	this.toneMappingExposure = 1.0;
 	this.toneMappingWhitePoint = 1.0;
 
+	this.passCamera = null;
+	this.passQuad = null;
+	this.passScene = null;
```
Comment (dimarudol) [inline src/renderers/WebGLRenderer.js]: There are more post/pass :)

Comment (bhouston) [inline src/renderers/WebGLRenderer.js]: I've settled on using pass* rather than post*, thanks for catching the inconsistencies, I appreicate it. :)

**Comment by bhouston** [comment]
Superceded by simplier PR: https://github.com/mrdoob/three.js/pull/8913

**Pull request closed without merging**